### PR TITLE
Replace Controller with Activity for ConversationList

### LINKED ISSUE #2615: Avatars gone in conversation list after screen rotation (...)
If you switch the theme ( regardless from light to dark, dark to light, in Talk or system settings) the avatars in the conversation list are gone:

If you run the app in debug, Coil will write logs. It looks like the the image loading actions will be canceled in the case of a theme switch:

```
2022-12-05 14:29:02.705 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - https://cloud.example.com/index.php/avatar/tim2/512
2022-12-05 14:29:02.729 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@f9c79c1
2022-12-05 14:29:02.734 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@d4cf954
2022-12-05 14:29:02.773 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@a2e8ed9
2022-12-05 14:29:02.826 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@43d1af1
2022-12-05 14:29:02.843 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@8d00a47
2022-12-05 14:29:02.858 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@9b0e0d
2022-12-05 14:29:02.876 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🏗  Cancelled - android.graphics.drawable.LayerDrawable@9be2fc3
```

In comparison the successful loading (during switching back to the list from a single conversation):

```2022-12-05 14:26:41.907 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY_CACHE) - https://cloud.example.com/index.php/avatar/tim/512
2022-12-05 14:26:41.972 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@2987616
2022-12-05 14:26:41.973 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@9e92b97
2022-12-05 14:26:41.973 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@159c184
2022-12-05 14:26:41.973 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@c45cd6d
2022-12-05 14:26:41.973 30511-30511 RealImageLoader         com.nextcloud.talk2                  I  🧠 Successful (MEMORY) - android.graphics.drawable.LayerDrawable@d9ddda2
```

_Originally posted by @timkrueger in https://github.com/nextcloud/talk-android/issues/2449#issuecomment-1339014297_

### PR DESCRIPTION
**Pull request by mahibi**
this PR replaces controller with Activity, see https://github.com/nextcloud/talk-android/issues/1551

also fix  #2615

### 🚧 TODO

This is the current state before my vacation (i'm back at 17.4.).
There are some minor obvious bugs known UI/theming-wise (fyi @AndyScherzinger if you would like to fix anything)

Other than that i added some workarounds that the app will work with the remaining controllers. 
From my pov everything should work, altough many things can be improved.

Remaining controllers are:
* ServerSelectionController
* WebViewLoginController
* AccountVerificationController
* SwitchAccountController
* LockedController

When the PR has been tested and found to be OK, feel free to merge @AndyScherzinger 

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not needed
- [x] 🔖 Capability is checked or not needed 
- [x] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [x] 📅 Milestone is set
- [x] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

### CODE REVIEW

File: app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt [hunk 1]
```diff
@@ -1216,53 +1269,47 @@ class ConversationsListController(bundle: Bundle) :
...
+                    WorkManager.getInstance().enqueue(accountRemovalWork)
+                    if (otherUserExists) {
+                        finish()
+                        startActivity(intent)
+                    } else if (!otherUserExists) {
+                        Log.d(TAG, "No other users found. AccountRemovalWorker will restart the app.")
                     }
+                }
+                .setNegativeButton(R.string.nc_settings_reauthorize) { _, _ ->
+                    // TODO
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1297]: todo

File: app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt [hunk 2]
```diff
@@ -1279,25 +1326,20 @@ class ConversationsListController(bundle: Bundle) :
...
-                                .pushChangeHandler(VerticalChangeHandler())
-                                .popChangeHandler(VerticalChangeHandler())
-                        )
+                        restartApp(this)
                     }
                 }
                 .setNegativeButton(R.string.nc_cancel) { _, _ ->
                     if (userManager.users.blockingGet().isNotEmpty()) {
-                        router.pushController(RouterTransaction.with(SwitchAccountController()))
+                        // TODO
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/conversationlist/ConversationsListActivity.kt:1338]: todo

File: app/src/main/java/com/nextcloud/talk/ui/dialog/ChooseAccountDialogFragment.java [hunk 3]
```diff
@@ -178,16 +180,18 @@ private void setupListeners(User user) {
...
-            });
-            binding.manageSettings.setOnClickListener(v -> {
-                dismiss();
-                ((MainActivity) getActivity()).openSettings();
-            });
-        }
+
+        binding.addAccount.setOnClickListener(v -> {
+            dismiss();
+            // TODO
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/ui/dialog/ChooseAccountDialogFragment.java:186]: todo! 
Blocker for merging...

File: app/src/main/java/com/nextcloud/talk/controllers/base/BaseController.kt [hunk 4]
```diff
@@ -54,13 +49,12 @@ import com.nextcloud.talk.controllers.ServerSelectionController
...
 import com.nextcloud.talk.controllers.WebViewLoginController
 import com.nextcloud.talk.controllers.base.providers.ActionBarProvider
-import com.nextcloud.talk.databinding.ActivityMainBinding
 import com.nextcloud.talk.ui.theme.ViewThemeUtils
-import com.nextcloud.talk.utils.DisplayUtils
 import com.nextcloud.talk.utils.preferences.AppPreferences
 import javax.inject.Inject
 import kotlin.jvm.internal.Intrinsics
 
+// TODO: check what needs to be migrated from this class to BaseActivity etc when conductor is removed
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/controllers/base/BaseController.kt:57]: todo. Maybe in follow up PR's..

**Review by AndyScherzinger — CHANGES_REQUESTED** [review]
Theming is fully broken regarding respecting server-side theming, coloring is always blue atm, needs to be fixed first before merging

**Review by AndyScherzinger — APPROVED** [review]

**Pull request merged**